### PR TITLE
docs: clarify dynamic completion features

### PR DESCRIPTION
**Pull request by senekor**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: docs/install-and-setup.md [hunk 1]
```diff
@@ -210,13 +210,17 @@ To set up command-line completion, source the output of
 `jj util completion bash/zsh/fish`. Exactly how to source it
 depends on your shell.
 
-Improved completions are currently in the works, these will complete things like
-bookmark names as well. You can activate them with the alternative "dynamic"
+Improved completions are also available, they will complete things like
```
Comment (martinvonz) [inline docs/install-and-setup.md:213]: Nit: "they will ... " is a full sentence, so use full stop and capital letter

File: docs/install-and-setup.md [hunk 2]
```diff
@@ -210,13 +210,17 @@ To set up command-line completion, source the output of
...
+bookmarks, aliases, revisions, operations and files as well. They can be
```
Comment (martinvonz) [inline docs/install-and-setup.md:214]: It took me a few seconds to figure out what "as well" refers to. I assume it's "as well as what the static completions do". I'm not sure we need to say that. What do you think about just deleting "as well"? Oh, we already say further down that they also support the static completions.

**Review by martinvonz — APPROVED** [review]

**Comment by senekor** [comment]
Btw. I hard wrapped these lines because that seems to be the convention. I personally prefer [one sentence per line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line).

**Comment by martinvonz** [comment]
I had not heard of that convention until very recently. It was mentioned in a recent PR for the tutorial (I'll let you search for it). As I said there, I would be fine with changing to that convention. Whichever convention we choose, we should usually find a formatter that can automate it for us.

**Pull request merged**